bql/semantic/semantic.go

```diff
@@ -75,6 +79,8 @@ type Statement struct {
 	data                      []*triple.Triple
 	pattern                   []*GraphClause
 	workingClause             *GraphClause
+	constructClauses          []*ConstructClause
```

Comment:
So `constructClauses` captures only the declarations within a `CONSTRUCT` statement that specifies how the triples must be built. For instance, consider the following:

`construct {?s "new_predicate"@[] ?o} into ?a from ?b where {?s "old_predicate"@[] ?o}`

`constructClauses` simply captures the `?s "new_predicate"@[] ?o` part. `workingClauses` will capture the `?s "old_predicate"@[] ?o` as before. `constructClauses` will capture multiple such declarations (separated by the `.` operator) and `constructClause` also has support for multiple reification clauses. So in a way, `constructClause` is similar to `data` within the `Statement` struct (except it has to support literals, bindings and reification). Does this seem like the right thing to do?